src/Types/AbstractTypedIDEntity.php

```diff
@@ -134,6 +141,14 @@ public function isValid()
         return \count($callbacks) === $passed_count;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function canAutodetect()
+    {
+        return $this->can_autodetect === true;
```

Comment:
Check boolean - is boolean? For a what?

Reply:
This property may be not boolean if user will be want to set its in extended class

Reply:
Nope. Make `return (bool) $this->can_autodetect`